GitUI/CommitInfo/CommitInfo.cs

```diff
@@ -224,6 +234,66 @@ private int GetRevisionHeaderHeight()
             return _headerResize.Height;
         }
 
+        private async Task ReloadCommitInfoAsync()
+        {
+            _RevisionHeader.BackColor = ColorHelper.MakeColorDarker(BackColor);
+
+            showContainedInBranchesToolStripMenuItem.Checked = AppSettings.CommitInfoShowContainedInBranchesLocal;
+            showContainedInBranchesRemoteToolStripMenuItem.Checked = AppSettings.CommitInfoShowContainedInBranchesRemote;
+            showContainedInBranchesRemoteIfNoLocalToolStripMenuItem.Checked = AppSettings.CommitInfoShowContainedInBranchesRemoteIfNoLocal;
+            showContainedInTagsToolStripMenuItem.Checked = AppSettings.CommitInfoShowContainedInTags;
+            showMessagesOfAnnotatedTagsToolStripMenuItem.Checked = AppSettings.ShowAnnotatedTagsMessages;
+
+            ResetTextAndImage();
+
+            if (string.IsNullOrEmpty(_revision.Guid))
+            {
+                Debug.Fail("Unexpectedly called ReloadCommitInfoAsync() with empty revision");
+                return;
+            }
+
+            _RevisionHeader.Text = string.Empty;
+            _RevisionHeader.Refresh();
+
+            CommitData data = _commitDataManager.CreateFromRevision(_revision);
+            data.ChildrenGuids = _children;
+
+            if (_revision.Body == null)
+            {
+                // TODO error response here is ignored
+                await _commitDataManager.UpdateCommitMessageAsync(data, _revision.Guid);
+                _revision.Body = data.Body;
+            }
+
+            ThreadPool.QueueUserWorkItem(_ => loadLinksForRevision(_revision));
+
+            if (_sortedRefs == null)
+                ThreadPool.QueueUserWorkItem(_ => loadSortedRefs());
+
+            var header = _commitDataHeaderRenderer.Render(data, CommandClick != null);
+            var body = _commitDataBodyRenderer.Render(data, CommandClick != null);
+
+            _RevisionHeader.SetXHTMLText(header);
+            _RevisionHeader.Height = _headerResize.Height;
+            _revisionInfo = body;
+
+            UpdateRevisionInfo();
+            LoadAuthorImage(data.Author ?? data.Committer);
+
+            //No branch/tag data for artificial commands
+            if (GitRevision.IsArtificial(_revision.Guid))
+                return;
+
+            if (AppSettings.CommitInfoShowContainedInBranches)
+                ThreadPool.QueueUserWorkItem(_ => loadBranchInfo(_revision.Guid));
```

Comment:
❗️ Asynchronous operation not captured in the lifetime of the enclosing operation must be explicitly tracked